src/Symfony/Component/Messenger/Transport/Receiver/ForceConsumptionReceiver.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Receiver;
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * @author Mathias STRASSER <contact@roukmoute.fr>
+ *
+ * @experimental in 4.2
+ */
+final class ForceConsumptionReceiver implements ReceiverInterface
+{
+    /** @var ReceiverInterface */
+    private $decoratedReceiver;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(ReceiverInterface $decoratedReceiver, LoggerInterface $logger = null)
+    {
+        $this->decoratedReceiver = $decoratedReceiver;
+        $this->logger = $logger;
+    }
+
+    public function receive(callable $handler): void
+    {
+        $this->decoratedReceiver->receive(
+            function ($message) use ($handler) {
```

Comment:
this is now accepting an evelope. it's missing typehint and $message should be renamed, see other receivers